PeripheryFunctions/TS_GetIDs.m

```diff
@@ -67,29 +73,27 @@
 case 'ops'
     theDataTable = Operations;
 otherwise
-    error('Specify ''ts'' or ''ops''');
+    error('Specify ''ts'', ''ops'', or ''opsName''');
 end
 
-% The cell of comma-delimited keyword strings:
-theKeywordCell = theDataTable.Keywords;
-
-% Split into sub-cells using comma delimiter:
-Keywords = SUB_cell2cellcell(theKeywordCell);
-
-% Find objects with a keyword that matchees that given:
-matches = cellfun(@(x)any(ismember(theKeyword,x)),Keywords);
+% OC: below has been changed so that we return a set in the same order as
+%       above, and nan's otherwise
+matches = nan(length(theFields),1);
+for i = 1:length(theFields)
+    cmatch = find(strcmp(theDataTable.(nameOrKeywords),theFields{i}));
+    if ~isempty(cmatch)
+        matches(i) = cmatch;
+    end
+end
 
-% Return the IDs of the matches:
-IDs = theDataTable.ID(matches);
+foundIDs = ~isnan(matches);
 
-% Check for empty:
-if isempty(IDs)
-    warning('No matches to ''%s'' found in %s',theKeyword,theDataFile)
-end
+IDs = zeros(size(matches));
+IDs(foundIDs) = theDataTable.ID(matches(foundIDs));
+IDs(~foundIDs) = nan;
```

Comment:
Why would you prefer this format to a list of IDs? If it's going to be like this, it could just be a binary label (true/false) of the same size.

Reply:
Ideally we would be able to know which input corresponds to which output, rather than simply whether or not it exists in the table. So we should be able to input a (sorted) list of names/IDs and get out which ID in the table they correspond to in the same order

Reply:
This function was aimed at getting IDs for those elements that are labeled with a specific keyword. Maybe you can clearly describe the functionality you are hoping to implement and where it would be useful? It's possible what you're trying to do (inputting a structure and getting a match for each) is a quite different thing that can either be a flag in the input to this function, or a new function altogether.

Reply:
The idea is to input a cell array of Names and get output a sorted list of where they are in the list. For instance, you might want to input the list of Catch-22 operations and get output what index they correspond to in HCTSA. Equivalently for time series, you might want the IDs of specific time-series names

Reply:
Ah ok yeah I understand now—let's integrate and make it a new toggleable mode of `TS_GetIDs`.